tests/tt_metal/tt_metal/unit_tests_fast_dispatch/command_queue/test_EnqueueProgram.cpp

```diff
@@ -1416,4 +1389,326 @@ TEST_F(CommandQueueFixture, TestRandomizedProgram) {
+            max_runtime_microseconds = MAX_KERNEL_RUNTIME_MICROSECONDS * (2.0 / 10);
+        }
+
+        this->create_kernel(
+            program,
+            CoreType::WORKER,
+            false,
+            min_num_sems,
+            max_num_sems,
+            min_num_cbs,
+            max_num_cbs,
+            min_num_rt_args,
+            max_num_rt_args,
+            min_size_bytes,
+            max_size_bytes,
+            min_runtime_microseconds,
+            max_runtime_microseconds);
+
+        EnqueueProgram(device_->command_queue(), program, false);
+    }
+
+    Finish(device_->command_queue());
+}
+
+TEST_F(RandomProgramFixture, TestLargeProgramInBetweenFiveSmallProgramsOnTensix) {
+    for (uint32_t i = 0; i < NUM_PROGRAMS; i++) {
+        if (i % 10 == 0) {
+            log_info(tt::LogTest, "Creating Program {}", i);
+        }
+        Program program = CreateProgram();
+
+        uint32_t min_num_sems;
+        uint32_t max_num_sems;
+        uint32_t min_num_cbs;
+        uint32_t max_num_cbs;
+        uint32_t min_num_rt_args;
+        uint32_t max_num_rt_args;
+        uint32_t min_size_bytes;
+        uint32_t max_size_bytes;
+        uint32_t min_runtime_microseconds;
+        uint32_t max_runtime_microseconds;
+        if (i % 6 == 0) {
+            min_num_sems = MAX_NUM_SEMS * (8.0 / 10);
+            max_num_sems = MAX_NUM_SEMS;
+            min_num_cbs = MAX_NUM_CBS * (8.0 / 10);
+            max_num_cbs = MAX_NUM_CBS;
+            min_num_rt_args = MAX_NUM_RUNTIME_ARGS * (9.0 / 10);
+            max_num_rt_args = MAX_NUM_RUNTIME_ARGS;
+            min_size_bytes = MAX_KERNEL_SIZE_BYTES * (9.0 / 10);
+            max_size_bytes = MAX_KERNEL_SIZE_BYTES;
+            min_runtime_microseconds = MAX_KERNEL_RUNTIME_MICROSECONDS * (9.0 / 10);
+            max_runtime_microseconds = MAX_KERNEL_RUNTIME_MICROSECONDS;
+        } else {
+            min_num_sems = MIN_NUM_SEMS;
+            max_num_sems = MAX_NUM_SEMS * (3.0 / 10);
+            min_num_cbs = MIN_NUM_CBS;
+            max_num_cbs = MAX_NUM_CBS * (3.0 / 10);
+            min_num_rt_args = max_num_sems + max_num_cbs;
+            max_num_rt_args = MAX_NUM_RUNTIME_ARGS * (3.0 / 10);
+            min_size_bytes = MIN_KERNEL_SIZE_BYTES;
+            max_size_bytes = MAX_KERNEL_SIZE_BYTES * (2.0 / 10);
+            min_runtime_microseconds = MIN_KERNEL_RUNTIME_MICROSECONDS;
+            max_runtime_microseconds = MAX_KERNEL_RUNTIME_MICROSECONDS * (2.0 / 10);
+        }
+
+        this->create_kernel(
+            program,
+            CoreType::WORKER,
+            false,
+            min_num_sems,
+            max_num_sems,
+            min_num_cbs,
+            max_num_cbs,
+            min_num_rt_args,
+            max_num_rt_args,
+            min_size_bytes,
+            max_size_bytes,
+            min_runtime_microseconds,
+            max_runtime_microseconds);
+
+        EnqueueProgram(device_->command_queue(), program, false);
```

Comment:
Add a Finish call here? Or make the EnqueueProgram blocking=true, otherwise you're relying on Device close to sync your test. Might lead to unexpected behaviour